### PR TITLE
feat(snippet): add example for @dbquery with SQL queries

### PR DESCRIPTION
**Pull request by aswanyaugustine1992**
This pull request introduces a new snippet that demonstrates the usage of the @dbquery(query:) directive in StepZen. This snippet is designed to help understand how to perform custom SQL queries using the @dbquery directive in a demo database.

### CODE REVIEW

File: dbquery/custom-query/api.graphql [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# This example shows how @dbquery is configured for custom SQL queries with Customer data.
+
+"""
+Sample Customer type.
```
Comment (sreekanthibm) [inline dbquery/custom-query/api.graphql:4]: Please add the table structure with column name, type etc..

File: dbquery/custom-query/api.graphql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+type Query {
+    # Fetches all customers.
+  getAllCustomers: [Customer]
+    @dbquery(
+      query: "SELECT id, name, email FROM customer"
+      type: "postgresql"
+      configuration: "postgresql_config"
+    )
+  # Fetches a customer by their ID.
+getCustomerById(id: Int!): Customer
```
Comment (sreekanthibm) [inline dbquery/custom-query/api.graphql:25]: Format issue

File: dbquery/custom-query/tests/Test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  const tests = [
+    {
+      label: "fetch all customers",
+      query: requests,
+      operationName: "GetAllCustomers",
+      variables: {},
+      expected: {
+        getAllCustomers: [
+          { id: "1", name: "Lucas Bill                                        ", email: "lucas.bill@example.com                            " },
```
Comment (sreekanthibm) [inline dbquery/custom-query/tests/Test.js:25]: Why empty space is coming for name and email

**Review by sreekanthibm — COMMENTED** [review]
Please address the suggested changes

File: dbquery/custom-query/tests/Test.js (hunk 3 above)
Comment (aswanyaugustine1992) [reply dbquery/custom-query/tests/Test.js:25]: Its how in the DB data resides

File: dbquery/custom-query/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# @dbquery SQL query Snippet
+
+This example demonstrates how to use the `@dbquery` directive with custom SQL queries for `Customer` data in StepZen.
+
+## Files
+
+- **api.graphql**: Contains the GraphQL schema for querying `Customer` objects using custom SQL queries.
+- **config.yaml**: Database configuration, including connection URI.
+- **index.graphql**: Entry point schema file.
+- **operations.graphql**: Sample queries to test the `@dbquery` directive.
```
Comment (ddebrunner) [inline dbquery/custom-query/README.md:10]: Sample **operations**

File: dbquery/custom-query/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+## Prerequisites
+
+- A PostgreSQL database with a `customer` table containing `id`, `name`, and `email` fields.
+- Replace the `uri` in `config.yaml` with your database connection details.
+
+## Usage
+
+1. Deploy the schema using StepZen.
+2. Use the sample queries in `operations.graphql` to test the `@dbquery` functionality.
```
Comment (ddebrunner) [inline dbquery/custom-query/README.md:21]: Sample operations

Comment (ddebrunner) [reply dbquery/custom-query/README.md:21]: Show how the operations can be executed

File: dbquery/custom-query/config.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+configurationset:
+  - configuration:
+      name: postgresql_config
+      uri: postgresql://postgresql.introspection.stepzen.net/introspection?user=testUserIntrospection&password=HurricaneStartingSample1934
```
Comment (ddebrunner) [inline dbquery/custom-query/config.yaml:4]: add missing newline

File: dbquery/custom-query/operations.graphql [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Query to get a customer by ID
```
Comment (ddebrunner) [inline dbquery/custom-query/operations.graphql:1]: These are operations

File: dbquery/custom-query/api.graphql [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+Table structure for 'customer':
+
+  CREATE TABLE customer (
+    id SERIAL PRIMARY KEY,          -- Unique identifier with sequence
+    name CHARACTER(50) NOT NULL,    -- Customer's name, max 50 characters
+    email CHARACTER(50) NOT NULL,   -- Customer's email, max 50 characters, must be unique
+    CONSTRAINT customer_email_key UNIQUE (email)  -- Unique constraint on email
+  );
+
+  Foreign key references:
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:25]: Are these relevant?

File: dbquery/custom-query/api.graphql [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  # Fetches a customer by their ID.
+  getCustomerById(id: Int!): Customer
+    @dbquery(
+      query: "SELECT id, name, email FROM customer WHERE id = $1",
+      type: "postgresql",
+      configuration: "postgresql_config"
+    )
+
+  # Searches customers by name using a pattern match.
+  searchCustomersByName(name: String!): [Customer]
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:47]: Does this add any value  over `getCustomerById` ?

File: dbquery/custom-query/api.graphql [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    CONSTRAINT customer_email_key UNIQUE (email)  -- Unique constraint on email
+  );
+
+  Foreign key references:
+    - customeraddress(customerid) references customer(id)
+    - "order"(customerid) references customer(id)
+"""
+type Query {
+  # Fetches all customers.
+  getAllCustomers: [Customer]
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:31]: Rename `customers`

File: dbquery/custom-query/api.graphql [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    @dbquery(
+      query: "SELECT id, name, email FROM customer",
+      type: "postgresql",
+      configuration: "postgresql_config"
+    )
+
+  # Fetches a customer by their ID.
+  getCustomerById(id: Int!): Customer
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:39]: Rename `customer`

File: dbquery/custom-query/api.graphql (hunk 8 above)
Comment (aswanyaugustine1992) [reply dbquery/custom-query/api.graphql]: Agree. These are not relevant.

File: dbquery/custom-query/tests/Test.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+          { id: "9", name: "Asha Kumari                                       ", email: "asha.kumari@example.com                           " },
+          { id: "10", name: "Salma Khan                                        ", email: "salma.khan@example.com                            " }
+        ]
+      },
+    },
+    {
+      label: "fetch customer by ID",
+      query: requests,
+      operationName: "Customer",
+      variables: {
```
Comment (ddebrunner) [inline dbquery/custom-query/tests/Test.js:42]: Why are variables used here?

File: dbquery/custom-query/config.yaml [hunk 13]
```diff
@@ -0,0 +1,5 @@
...
+
```
Comment (ddebrunner) [inline dbquery/custom-query/config.yaml:5]: Newlines!

File: dbquery/custom-query/api.graphql [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+Sample Customer type.
+"""
+type Customer {
+  id: ID!
+  name: String
+  email: String
+}
+
+"""
+`Customer` queries a mock `Customer` object from a demo PostgreSQL database.
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:13]: What is `Customer` ?

File: dbquery/custom-query/api.graphql [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  # Fetches all customers.
+  customers: [Customer]
+    @dbquery(
+      query: "SELECT id, name, email FROM customer",
+      type: "postgresql",
+      configuration: "postgresql_config"
+    )
+
+  # Fetches a customer by their ID.
+  customer(id: Int!): Customer
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:36]: Expand comments here to describe the mapping between field arguments and parameter markers. 
Snippets are a teaching aid, so it helps to describe details of  the directive usage.

File: dbquery/custom-query/operations.graphql [hunk 16]
```diff
@@ -0,0 +1,17 @@
+# Retrieve a specific customer by ID
+query Customer {
+  customer(id: 1) {
```
Comment (ddebrunner) [inline dbquery/custom-query/operations.graphql:3]: Maybe use variables?

File: dbquery/custom-query/README.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+  type: "postgresql",
+  configuration: "your_config"
+)
+```
+
+This allows you to fetch exactly the data you need, based on the custom SQL query provided. You can adjust the queries to match your database schema and use case.
+
+## Try it Out!
+
+Deploy the schema from `dbquery/pagination` relative to the repository's root directory:
```
Comment (ddebrunner) [inline dbquery/custom-query/README.md:48]: Fix path name

File: dbquery/custom-query/api.graphql [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+    Fetches a list of all customers from the database.
+
+    The custom SQL query retrieves the `id`, `name`, and `email` fields from the 'customer' table.
+
+    **@dbquery Directive Usage**:
+    - `query`: This is the SQL query that will be executed. Here, it fetches all records from the 'customer' table.
+    - `type`: Specifies the type of database being queried. In this case, it’s a PostgreSQL database.
+    - `configuration`: References the database configuration (connection details) in StepZen.
+
+    This query does not take any arguments, and hence there are no parameter markers in the SQL query.
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:40]: field does not take any arguments

File: dbquery/custom-query/api.graphql [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+            type: "postgresql"
+            configuration: "postgresql_config"
+        )
+
+    """
+    Fetches a single customer by their ID from the database.
+
+    **Field Argument to Parameter Marker Mapping**:
+    - The `id` argument provided in the GraphQL query maps to the `$1` marker in the SQL query.
+    - SQL queries typically use numbered markers (e.g., `$1`, `$2`, etc.) for parameterized inputs. The number refers to the position of the argument in the query.
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:55]: Actually they typically just use `?`, only postgres uses `$n`

File: dbquery/custom-query/api.graphql [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+    - When a query is executed, the value passed to the `id` argument in GraphQL (via variables) will replace the `$1` marker in the SQL query.
+
+    **@dbquery Directive Usage**:
+    - `query`: This is the SQL query that will be executed. Here, the customer data is fetched based on the provided `id` value.
+      - The `$1` marker is a placeholder for the value of the `id` argument, which is passed as a variable when executing the query.
+    - `type`: Specifies the type of database being queried (PostgreSQL).
+    - `configuration`: References the database configuration (connection details) in StepZen.
+
+    This query requires an `id` argument as input, which is passed as a variable from the GraphQL query. The variable's value is used to fetch the corresponding customer from the database.
```
Comment (ddebrunner) [inline dbquery/custom-query/api.graphql:64]: This field ...
GraphQL query --> GraphQL request

**Review by ddebrunner — COMMENTED** [review]
Minor changes

**Review by ddebrunner — APPROVED** [review]
merge and do cleanup later.

**Pull request merged**